### PR TITLE
feat: use @ant-design/icons@4.0

### PR DESCRIPTION
**Pull request by vagusX**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [x] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/issues/16911

### 💡 Background and solution

use @ant-design/icons v4.0 to replace the icon solution in antd v3

### 📝 Changelog

not yet

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

**Comment by vagusX** [comment]
Missing icons in [iconfont.cn Ant Design 官方图标库](https://www.iconfont.cn/collections/detail?cid=9402)

* exclamation-circle-fill

File: components/badge/demo/dot.md [hunk 1]
```diff
@@ -14,15 +14,16 @@ title:
...
 
 ReactDOM.render(
   <div>
     <Badge dot>
-      <Icon type="notification" />
+      <Notification />
     </Badge>
     <Badge count={0} dot>
-      <Icon type="notification" />
+      <Notification />
```
Comment (afc163) [inline components/badge/demo/dot.md]: 这样很容易和 `import { Notification } from 'antd';` 命名冲突。

Comment (vagusX) [reply components/badge/demo/dot.md]: 这个图标名称是需要再考虑下，比如 Upload 这种就比较奇怪

Comment (afc163) [reply components/badge/demo/dot.md]: 统一加 Icon 前缀或后缀如何？

Comment (vagusX) [reply components/badge/demo/dot.md]: 可以的，感觉后缀更好些

Comment (zombieJ) [reply components/badge/demo/dot.md]: 让用户自己别名吧，库里保持简单

File: package.json [hunk 2]
```diff
@@ -104,7 +104,7 @@
     "@sentry/browser": "^5.4.0",
     "@types/classnames": "^2.2.8",
     "@types/prop-types": "^15.7.1",
-    "@types/react": "~16.8.19",
+    "@types/react": "~16.9.0",
```
Comment (afc163) [inline package.json]: 合并 feature，不然又要冲突了。

Comment (vagusX) [reply package.json]: 等这个分支合并后在 4.0-prepare 上合并哈

**Comment by vagusX** [comment]
close infavor of https://github.com/ant-design/ant-design/pull/18217

**Pull request closed without merging**